pepdbagent/modules/annotation.py

```diff
@@ -0,0 +1,341 @@
+                namespace=namespace, search_str=query, admin=admin
+            ),
+            result=self._get_projects(
+                namespace=namespace,
+                search_str=query,
+                admin=admin,
+                offset=offset,
+                limit=limit,
+            ),
+        )
+
+    def get_by_rp(
+        self,
+        registry_paths: Union[List[str], str],
+        admin: Union[List[str], str] = None,
+    ) -> AnnotationReturnModel:
+        """
+        Get project annotations by providing registry_path or list of registry paths.
+        :param registry_paths: registry path string or list of registry paths
+        :param admin: list of namespaces where user is admin
+                :return: pydantic model: AnnotationReturnModel(
+            limit:
+            offset:
+            count:
+            result: List [ AnnotationModel(
+                        namespace:
+                        name:
+                        tag:
+                        is_private:
+                        number_of_samples:
+                        description:
+                        last_update_date:
+                        submission_date:
+                        digest:
+                    )
+                ]
+            )
+        """
+        if isinstance(registry_paths, list):
+            anno_results = []
+            for path in registry_paths:
+                try:
+                    namespace, name, tag = registry_path_converter(path)
+                except RegistryPathError as err:
+                    _LOGGER.error(str(err), registry_paths)
+                    continue
+                single_return = self._get_single_annotation(namespace, name, tag, admin)
+                if single_return:
+                    anno_results.append(single_return)
+            return_len = len(anno_results)
+            return AnnotationReturnModel(
+                count=return_len,
+                limit=return_len,
+                offset=len(registry_paths),
+                result=anno_results,
+            )
+
+        else:
+            try:
+                namespace, name, tag = registry_path_converter(registry_paths)
+            except RegistryPathError as err:
+                _LOGGER.error(str(err), registry_paths)
+                return AnnotationReturnModel()
+            return self.get(namespace=namespace, name=name, tag=tag, admin=admin)
+
+    def _get_single_annotation(
+        self,
+        namespace: str = None,
+        name: str = None,
+        tag: str = None,
+        admin: Union[List[str], str] = None,
+    ) -> Union[AnnotationModel, None]:
+        """
+        Retrieving project annotation dict by specifying project name
+        :param namespace: project registry_path - will return dict of project annotations
+        :param name: project name in database
+        :param tag: tag of the projects
+        :param admin: list of namespaces where user is admin
```

Comment:
Is there a proper way to indicate multiple possible types in the docstring?

Reply:
Do we use sphinx for this? I think you can link with `or`. From the [docs](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#info-field-lists)

Reply:
I was using basic restructuredtext docstring style, but without specifying return value 😬